BlueDB/src/main/java/org/bluedb/api/BlueCollectionBuilder.java

```diff
@@ -4,8 +4,10 @@
 import java.util.Collection;
 
 import org.bluedb.api.exceptions.BlueDbException;
+import org.bluedb.api.keys.BlueKey;
 
-public interface BlueCollectionBuilder<T extends Serializable> {
-	public BlueCollectionBuilder<T> withOptimizedClasses(Collection<Class<? extends Serializable>> classesToRegister);
+public interface BlueCollectionBuilder<K extends BlueKey, T extends Serializable> {
+	public BlueCollectionBuilder<K, T> withOptimizedClasses(Collection<Class<? extends Serializable>> classesToRegister);
+	public BlueCollectionBuilder<K, T> withRequestedSegmentSize(SegmentSize<K> segmentSize);

```

Comment:
I don't love the word requested. It feels like bluedb could just decide to ignore you. In reality, bluedb will absolutely take it if its a new collection but won't if the collection already exists and has a different segment size. When you copy files in the Java Files API it has you pass in an optional parameter that allows you to state what to do if the files already exist. For example, you can overwrite all or skip duplicates. We could possible allow them to pass in an enum that tells us what to do. For now there would only be one option which would be the default and which would just use the existing segment size. In the future if we added better support for switching between segment sizes then we could add a migrate option without breaking the API.

Reply:
Are you suggesting any change right now or just talking about possibly for later?